meson.build

```
@@ -3,8 +3,10 @@ project(
 	'substrate',
 	'cpp',
 	default_options: [
-		'cpp_std=c++11',
-		'warning_level=3'
+		'cpp_std=c++14',
+		'warning_level=3',
+		'buildtype=debugoptimized',
+		'default_library=static',
```

As discussed with you directly, these are the only two lines in this patch set we have any problems with. You've explained to us that the change to static is to catch some symbol visibility issues, and we think a better choice would be to add a CI job for static+LTO on macOS instead.

Our suggestion here is:
```suggestion
		'buildtype=release',
		'default_library=shared',
		'b_ndebug=if-release',
		'b_lto=true',
```

which also gets the defaults we would apply written down and out our head. Please let us know what you think of this compromise and if you think we're making a mistake.